docs/source/hooks/common_use_cases.md

```diff
@@ -0,0 +1,115 @@
+# Common use cases
+
+## Use Hooks to extend a node's behaviour
+
+You can add extra behaviour before and after a node's execution by using the [`before_node_run` and `after_node_run` Hooks](/kedro.framework.hooks.specs.NodeSpecs). Furthermore, you can apply extra behaviour to not only an individual node or an entire Kedro pipeline, but also to a _subset_ of nodes based on their tags or namespaces. For example, let's say we want to add the following extra behaviours to a node:
+
+```python
+from kedro.pipeline.node import Node
+
+
+def say_hello(node: Node):
+    """An extra behaviour for a node to say hello before running."""
+    print(f"Hello from {node.name}")
+```
+
+Then you can either add it to a single node based on the node's name:
+
+```python
+# <your_project>/src/<your_project>/hooks.py
+
+from kedro.framework.hooks import hook_impl
+from kedro.pipeline.node import Node
+
+
+class ProjectHooks:
+    @hook_impl
+    def before_node_run(self, node: Node):
+        # adding extra behaviour to a single node
+        if node.name == "hello":
+            say_hello(node)
+```
+
+Or add it to a group of nodes based on their tags:
+
+
+```python
+# <your_project>/src/<your_project>/hooks.py
+
+from kedro.framework.hooks import hook_impl
+from kedro.pipeline.node import Node
+
+
+class ProjectHooks:
+    @hook_impl
+    def before_node_run(self, node: Node):
+        if "hello" in node.tags:
+            say_hello(node)
+```
+
+Or add it to all nodes in the entire pipeline:
+
+```python
+# <your_project>/src/<your_project>/hooks.py
+
+from kedro.framework.hooks import hook_impl
+from kedro.pipeline.node import Node
+
+
+class ProjectHooks:
+    @hook_impl
+    def before_node_run(self, node: Node):
+        # adding extra behaviour to all nodes in the pipeline
+        say_hello(node)
+```
+
+If your use case takes advantage of a decorator, for example to retry a node's execution using a library such as [tenacity](https://tenacity.readthedocs.io/en/latest/), you can still decorate the node's function directly:
+
+```python
+from tenacity import retry
+
+
+@retry
+def my_flaky_node_function():
+    ...
+```
+
+Or applying it in the `before_node_run` Hook as follows:
+
+```python
+# <your_project>/src/<your_project>/hooks.py
+from tenacity import retry
+
+from kedro.framework.hooks import hook_impl
+from kedro.pipeline.node import Node
+
+
+class ProjectHooks:
+    @hook_impl
+    def before_node_run(self, node: Node):
+        # adding retrying behaviour to nodes tagged as flaky
+        if "flaky" in node.tags:
+            node.func = retry(node.func)
+```
+## Use Hooks to customise the dataset load and save methods
```

Comment:
I suggest splitting this into a separate file - this one is already long enough.

Reply:
Wouldn't that leave the `Use Hooks to customise the dataset load and save methods` page quite short?

Reply:
About the length of a typical screen would be the ideal - rather short than long, as long as it makes sense. But this isn't a hard and fast rule, so I'm not going to shout about it!

Reply:
I'm okay with the current page length. This works!